### PR TITLE
Code to process phase2 version of deepTauID v2p5 [12_5_X]

### PR DESCRIPTION
**Pull request by mbluj**
#### PR description:

This PR adds code for a phase-2 version of DeepTau discriminator; it is a backport of https://github.com/cms-sw/cmssw/pull/40622.
The neural network structure of the new phase-2 version is identical to that of Run-2/3 DeepTau v2p5, therefore code modifications are small. The changes include:
* Adding scaling parameters and working points for the new training;
* A new "year" identifier introduced to separate this new phase-2 training and current Run-2/3 v2p5 (2026 vs 2018);
* The electron input collection used for the new phase2 deepTauID is sum of two collections: `slimmedElectrons` (gsfElectrons in EB) and `slimmedElectronsHGC` (HGCal electrons in endcaps). Notes: 1. The latter collection is called `slimmedElectronsFromMultiCl` in older CMSSW release series (and samples produced with them) which can cause some problems with running on old samples; 2. The same merged electron collection is used by old-style anti-electron tau discriminant for phase-2;

The performance of this new phase-2 DeepTau discriminator is documented in [AN-22-090](https://icms.cern.ch/tools-api/restplus/relay/piggyback/notes/AN/2022/90/files/3/download).

The new data-model was merged to cms-data in the following PR: https://github.com/cms-data/RecoTauTag-TrainingFiles/pull/11

This PR allows to run the new version of DeepTau discriminator in private workflows (to analyse phase-2 samples with 12_5_X), but **it does not enable the new version of DeepTau in official workflows**.  Therefore, differences in outputs are not expected.

To run standalone test, one can run the configuration file `RecoTauTag/RecoTau/test/runDeepTauIDsOnMiniAOD.py` after setting the flag `phase2 = True` in that file.

#### PR validation:

Validated with the standalone test mentioned above and with miniAOD workflows.

#### Backport PR

This is backport of https://github.com/cms-sw/cmssw/pull/40622 to 12_5_X to facilitate tau studies for phase-2 with 12_5_X samples.

### CODE REVIEW

**Comment by mandrenguyen** [comment]
please test

**Comment by swertz** [comment]
+1

**Comment by mandrenguyen** [comment]
+1

**Comment by mbluj** [comment]
Just a comment: to make this PR useful also cms-dist of 12_5_X should be updated to contain cms-data state with data-model of the new phase-2 integrated in, i.e. with PR https://github.com/cms-data/RecoTauTag-TrainingFiles/pull/11

File: PhysicsTools/PatAlgos/python/slimming/miniAOD_tools.py [hunk 1]
```diff
@@ -382,8 +382,8 @@ def _add_deepFlavour(process):
         toKeep = ['deepTau2017v2p1','deepTau2018v2p5']
     )
     from Configuration.Eras.Modifier_phase2_common_cff import phase2_common #Phase2 Tau MVA
-    phase2_common.toModify(tauIdEmbedder.toKeep, func=lambda t:t.append('newDMPhase2v1')) #Phase2 Tau isolation MVA
-    phase2_common.toModify(tauIdEmbedder.toKeep, func=lambda t:t.append('againstElePhase2v1')) #Phase2 Tau anti-e MVA
+    _tauIds_phase2 = ['newDMPhase2v1','againstElePhase2v1']
```
Comment (perrotta) [inline PhysicsTools/PatAlgos/python/slimming/miniAOD_tools.py:385]: @mbluj `newDMPhase2v1` was not kept in the master version. Just to try to understand: why?

Comment (perrotta) [reply PhysicsTools/PatAlgos/python/slimming/miniAOD_tools.py:385]: > @mbluj `newDMPhase2v1` was not kept in the master version. Just to try to understand: why?

It has quite likely something to do with https://github.com/cms-sw/cmssw/pull/40724#issuecomment-1424301473, but I fail to find the connection (sorry...)

**Comment by perrotta** [comment]
please test with https://github.com/cms-sw/cmsdist/pull/8308

**Comment by perrotta** [comment]
merge

**Pull request merged**